lib/types.js

```diff
@@ -21,10 +21,11 @@ export type Message = {
   severity: 'error' | 'warning' | 'info',
   reference: ?string,
   solutions?: Array<{
-    title: ?string,
-    position: Range,
-    currentText: ?string,
-    replaceWith: string,
+    title?: string,
+    position?: Range,
+    currentText?: string,
+    replaceWith?: string,
+    apply: (() => void)
```

Comment:
You're marking the callback as _required_, and changing the `replaceWith` to optional, if anything they should both be optional, with a note in the documentation that one or the other (but not both?) is required.